src/CdrWriter.test.ts

```diff
@@ -56,57 +72,59 @@ describe("CdrWriter", () => {
   });
 
   it("round trips all data types", () => {
-    const writer = new CdrWriter();
-    writer.int8(-1);
-    writer.uint8(2);
-    writer.int16(-300);
-    writer.uint16(400);
-    writer.int32(-500_000);
-    writer.uint32(600_000);
-    writer.int64(-7_000_000_001n);
-    writer.uint64(8_000_000_003n);
-    writer.uint16BE(0x1234);
-    writer.uint32BE(0x12345678);
-    writer.uint64BE(0x123456789abcdef0n);
-    writer.float32(-9.14);
-    writer.float64(1.7976931348623158e100);
-    writer.string("abc");
-    writer.sequenceLength(42);
-    const data = writer.data;
-    expect(data.byteLength).toEqual(80);
-
-    const reader = new CdrReader(data);
-    expect(reader.int8()).toEqual(-1);
-    expect(reader.uint8()).toEqual(2);
-    expect(reader.int16()).toEqual(-300);
-    expect(reader.uint16()).toEqual(400);
-    expect(reader.int32()).toEqual(-500_000);
-    expect(reader.uint32()).toEqual(600_000);
-    expect(reader.int64()).toEqual(-7_000_000_001n);
-    expect(reader.uint64()).toEqual(8_000_000_003n);
-    expect(reader.uint16BE()).toEqual(0x1234);
-    expect(reader.uint32BE()).toEqual(0x12345678);
-    expect(reader.uint64BE()).toEqual(0x123456789abcdef0n);
-    expect(reader.float32()).toBeCloseTo(-9.14);
-    expect(reader.float64()).toBeCloseTo(1.7976931348623158e100);
-    expect(reader.string()).toEqual("abc");
-    expect(reader.sequenceLength()).toEqual(42);
+    for (const writer of [new CdrWriter(), new CdrWriter({ kind: EncapsulationKind.CDR2_LE })]) {
```

Comment:
can we make these tests with multiple writers use it.each? and is there any harm in going through all of the encapsulation kinds?